backend/gene_info/api/ensembl_ids.py

```diff
@@ -0,0 +1,73 @@
+import enum
+import gzip
+
+
+class SupportedOrganisms(enum.Enum):
+    HOMO_SAPIENS = "NCBITaxon:9606"
+    MUS_MUSCULUS = "NCBITaxon:10090"
+    SARS_COV_2 = "NCBITaxon:2697049"
+    ERCC = "NCBITaxon:32630"
+
+
+class GeneChecker:
+    """Handles checking gene ids, retrieves symbols"""
+
+    base_prefix = "backend/ontology_files/"
+    GENE_FILES = {
+        SupportedOrganisms.HOMO_SAPIENS: base_prefix + "genes_homo_sapiens.csv.gz",
+        SupportedOrganisms.MUS_MUSCULUS: base_prefix + "genes_mus_musculus.csv.gz",
+        SupportedOrganisms.SARS_COV_2: base_prefix + "genes_sars_cov_2.csv.gz",
+        SupportedOrganisms.ERCC: base_prefix + "genes_ercc.csv.gz",
+    }
+
+    gene_dict = {}
+
+    def __init__(self):
+
+        if not GeneChecker.gene_dict:
+            for file in self.GENE_FILES.keys():
+                with gzip.open(self.GENE_FILES[file], "rt") as genes:
+
+                    for gene in genes:
+
+                        gene = gene.rstrip().split(",")
+                        gene_id = gene[0]
+                        gene_label = gene[1]
+
+                        GeneChecker.gene_dict[gene_label] = gene_id
+
+    def is_valid_label(self, gene_label: str) -> bool:
+        """
+        Checks if gene name is in dictionary
+
+        :param str gene_label: human-readable gene symbol
+
+        :rtype bool
+        :return True if the gene_label is a valid gene name, False otherwise
+        """
+
+        return gene_label in GeneChecker.gene_dict
+
+    def get_id(self, gene_label) -> str:
+        """
+        Gets ENSEMBL id associated to the gene label
+
+        :param str gene_label: gene symbol
+
+        :rtype str
+        :return A gene ENSEMBL id
+        """
+
+        # in processing, if there are duplicate gene labels,
+        # a gene might be gene_label + "_" + gene_id
+        if "_" in gene_label:
+            gene_label = gene_label.split("_")[1]
+
+        if not self.is_valid_label(gene_label):
```

Comment:
you could probably rewrite this to be 
`if not gene_label in GeneChecker.gene_dict:` and remove the is_valid_label function for simplicity